cmd/cql-fuse/block_test.go

```diff
@@ -0,0 +1,540 @@
+/*
+ * Copyright 2018 The CovenantSQL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/CovenantSQL/CovenantSQL/client"
+	"github.com/CovenantSQL/CovenantSQL/utils"
+	"github.com/CovenantSQL/CovenantSQL/utils/log"
+)
+
+var (
+	baseDir        = utils.GetProjectSrcDir()
+	testWorkingDir = FJ(baseDir, "./test/")
+	logDir         = FJ(testWorkingDir, "./log/")
+	db             *sql.DB
+)
+
+var nodeCmds []*utils.CMD
+
+var FJ = filepath.Join
+
+func TestMain(m *testing.M) {
+	os.Exit(func() int {
+		var stop func()
+		db, stop = initTestDB()
+		defer stop()
+		defer db.Close()
+		return m.Run()
+	}())
+}
+
+func startNodes() {
+	ctx := context.Background()
+
+	// wait for ports to be available
+	var err error
+
+	err = utils.WaitForPorts(ctx, "127.0.0.1", []int{
+		6122,
+		6121,
+		6120,
+	}, time.Millisecond*200)
+
+	if err != nil {
+		log.Fatalf("wait for port ready timeout: %v", err)
+	}
+
+	// start 3bps
+	var cmd *utils.CMD
+	if cmd, err = utils.RunCommandNB(
+		FJ(baseDir, "./bin/cqld.test"),
+		[]string{"-config", FJ(testWorkingDir, "./fuse/node_0/config.yaml"),
+			"-test.coverprofile", FJ(baseDir, "./cmd/cql-minerd/leader.cover.out"),
```

Comment:
Should put this cover.out coverage result to cql-fuse directory to avoid write corruption with cql-minerd/integration_test.